integration-tests/light_load_all_functionality.py

```diff
@@ -27,8 +29,25 @@ def __str__(self):
         return repr(self.parameter)
 
 
+# range of ports where available ports can be found
+PORT_RANGE = [34256, 40000]
+
+
+def find_unbound_port():
+    """
+    Returns an unbound port number on 127.0.0.1.
+    """
+    port = random.randint(*PORT_RANGE)
+    sock = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
+    try:
+        sock.bind(("127.0.0.1", port))
+    except socket.error:
+        port = get_port()
```

Comment:
Yeah. Copied this from the internet. Changed it to a while loop instead of using recursion.